repo/repo.go

```diff
@@ -17,38 +19,57 @@ import (
 	"go.opentelemetry.io/otel"
 )
 
-type SignedCommit struct {
-	Root cid.Cid `cborgen:"root"`
-	Sig  []byte  `cborgen:"sig"`
-}
+// current version of repo currently implemented
+const ATP_REPO_VERSION int64 = 2
 
-type Root struct {
-	AuthToken *string  `cborgen:"auth_token"`
-	Data      cid.Cid  `cborgen:"data"`
-	Meta      cid.Cid  `cborgen:"meta"`
-	Prev      *cid.Cid `cborgen:"prev,omitempty"`
+type SignedCommit struct {
+	Did     string   `cborgen:"did"`
+	Version int64    `cborgen:"version"`
+	Prev    *cid.Cid `cborgen:"prev,omitempty"`
+	Data    cid.Cid  `cborgen:"data"`
+	Sig     []byte   `cborgen:"sig"`
 }
 
-type Meta struct {
-	Datastore string `cborgen:"datastore"`
-	Did       string `cborgen:"did"`
-	Version   int64  `cborgen:"version"`
+type UnsignedCommit struct {
+	Did     string   `cborgen:"did"`
+	Version int64    `cborgen:"version"`
+	Prev    *cid.Cid `cborgen:"prev,omitempty"`
+	Data    cid.Cid  `cborgen:"data"`
 }
 
 type Repo struct {
-	sr  SignedCommit
+	sc  SignedCommit
 	cst cbor.IpldStore
 	bs  blockstore.Blockstore
 
 	repoCid cid.Cid
 
-	meta *Meta
-
 	mst *mst.MerkleSearchTree
 
 	dirty bool
 }
 
+// Returns a copy of commit without the Sig field. Helpful when verifying signature.
+func (sc *SignedCommit) Unsigned() UnsignedCommit {
+	return UnsignedCommit{
+		Did:     sc.Did,
+		Version: sc.Version,
+		Prev:    sc.Prev,
+		Data:    sc.Data,
+	}
+}
+
+// returns SHA-256 (as direct bytes, not hex-encoded string bytes) of the
+// DAG-CBOR representation of object. This is what gets signed.
+func (uc *UnsignedCommit) BytesForSigning() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	if err := uc.MarshalCBOR(buf); err != nil {
+		return []byte{}, err
+	}
+	h := sha256.Sum256(buf.Bytes())
```

Comment:
I believe the signing library hashes the input already, need to verify

Reply:
Confirmed (go-did/key.go.(PrivKey).Sign) hashes the input with sha256 before signing internally.
I'm open to doing that elsewhere, but it seems less error prone there

Reply:
Ok, I will pull this hash out.